dnscache.go

```diff
@@ -6,22 +6,27 @@ import (
 	"sync"
 	"time"
 
-	"github.com/rs/dnscache/internal/singleflight"
+	"golang.org/x/sync/singleflight"
 )
 
+type DNSResolver interface {
+	LookupHost(ctx context.Context, host string) (addrs []string, err error)
+	LookupAddr(ctx context.Context, addr string) (names []string, err error)
+}
+
 type Resolver struct {
 	// Timeout defines the maximum allowed time allowed for a lookup.
 	Timeout time.Duration
 
 	// Resolver is the net.Resolver used to perform actual DNS lookup. If nil,
 	// net.DefaultResolver is used instead.
-	Resolver *net.Resolver
+	Resolver DNSResolver
 
 	once  sync.Once
 	mu    sync.RWMutex
 	cache map[string]*cacheEntry
 
-	onCacheMiss func()
+	OnCacheMiss func()
```

Comment:
If we expose this, we need a description of what it does.

Reply:
Of course, I missed the description of the method